src/main/java/net/snowflake/client/core/CredentialManager.java

```diff
@@ -114,24 +137,43 @@ synchronized void fillCachedCredential(SFLoginInput loginInput, String credType)
     }
 
     // cred can be null
-    if (credType == ID_TOKEN) {
-      logger.debug(
-          "Setting {}id token for user: {}, host: {}",
-          cred == null ? "null " : "",
-          loginInput.getUserName(),
-          loginInput.getHostFromServerUrl());
-      loginInput.setIdToken(cred);
-    } else if (credType == MFA_TOKEN) {
-      logger.debug(
-          "Setting {}mfa token for user: {}, host: {}",
-          cred == null ? "null " : "",
-          loginInput.getUserName(),
-          loginInput.getHostFromServerUrl());
-      loginInput.setMfaToken(cred);
-    } else {
-      logger.debug("Unrecognized type {} for local cached credential", credType);
+    switch (credType) {
+      case ID_TOKEN:
+        logger.debug(
```

Comment:
the messages are different only in the part after `{}` - could we make the login before and once?

Reply:
Done.